ouroboros-network/src/Ouroboros/Network/Point.hs

```diff
@@ -0,0 +1,29 @@
+module Ouroboros.Network.Point
+  ( WithOrigin (..)
+  , Block (..)
+  , origin
+  , at
+  , block
+  ) where
+
+data WithOrigin t = Origin | At t
+  deriving (Eq, Ord, Show)
+
+instance Functor WithOrigin where
+  fmap _ Origin    = Origin
+  fmap f (At t) = At (f t)
+
+data Block slot hash = Block
```

Comment:
What about calling this `BlockPoint`?

Reply:
In this module it is clear that it is a point since (apart from the type, the module is named `Point`), in most places we use unqualified imports so this might be confusing.  Or maybe just leave a comment at the top that this module is supposed to be imported qualified.

Reply:
I see now it is imported qualified in other places, just dropping a comment at the top might be useful.